### PR TITLE
feat:func to return Count of entities in registry table w/signoff

### PR DESCRIPTION
**Pull request by Jougan-0**
**Description**

This PR fixes #

**Notes for Reviewers**
This function returns the total count of entities in the registries table

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@VihasMakwana would you like to offer review on this PR?

File: models/meshmodel/registry/error.go [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
+		return
+	case v1alpha1.RelationshipDefinition:
+		NonImportModel.Relationships++
+		return
+	case v1alpha1.PolicyDefinition:
+		NonImportModel.Policies++
+		return
+	default:
+		return
+	}
```
Comment (VihasMakwana) [inline models/meshmodel/registry/error.go:44]: we can get rid of these return statements. It doesn't fallthrough unless specified.
```suggestion
func onEntityError(en Entity) {
	switch en.(type) {
	case v1alpha1.ComponentDefinition:
		NonImportModel.Components++
	case v1alpha1.RelationshipDefinition:
		NonImportModel.Relationships++
	case v1alpha1.PolicyDefinition:
		NonImportModel.Policies++
	}
```

File: models/meshmodel/registry/error.go [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
+	RegistryCount = make(map[string]int)
+}
+
 var (
 	ErrUnknownHostCode = "11097"
 )
 
 func ErrUnknownHost(err error) error {
 	return errors.New(ErrUnknownHostCode, errors.Alert, []string{"host is not supported"}, []string{err.Error()}, []string{"The component's host is not supported by the version of server you are running"}, []string{"Try upgrading to latest available version"})
 }
```
Comment (leecalcote) [inline models/meshmodel/registry/error.go:24]:
```suggestion
func ErrUnknownHost(err error) error {
	return errors.New(ErrUnknownHostCode, errors.Alert, []string{"Registrant type is not supported or unknown."}, []string{err.Error()}, []string{"The host registering a Model and it's components is not recognized by Meshery Server (or by the version currently running)."}, []string{"Validate the name and location of the model registrant. Try upgrading to latest available Meshery version."})
}
```

File: models/meshmodel/registry/registry.go [hunk 3]
```diff
@@ -440,6 +465,22 @@ func (rm *RegistryManager) GetRegistrant(e Entity) Host {
 	_ = rm.db.Where("id = ?", reg.RegistrantID).Find(&h).Error
 	return h
 }
+func (rm *RegistryManager) GetRegisteriesCount() (v1alpha1.EntitySummary, error) {
```
Comment (leecalcote) [inline models/meshmodel/registry/registry.go:468]: Please rename this function. There is only one Registry. There may be multiple Registrants (hosts that register Models, which contain Components).

File: models/meshmodel/registry/registry.go [hunk 4]
```diff
@@ -440,6 +465,22 @@ func (rm *RegistryManager) GetRegistrant(e Entity) Host {
...
+
+	var counts v1alpha1.EntitySummary
+	db := rm.db
+
+	query := db.Table("registries").
```
Comment (leecalcote) [inline models/meshmodel/registry/registry.go:473]: Please confirm that these counts are both per Registrant and aggregate total for the whole Registry.

**Review by leecalcote — COMMENTED** [review]
One of the more important use cases to facilitate with these logs is that of when a model or any of its entities is not successful in registering. It's important to be able to give contributors and Meshery administrators as much information as possible about what model and what entity failed to import and why.

Please provide examples of how these logs facilitate the administrator's resolution in the face of these import failures.

Include a link to troubleshooting Docs?

**Comment by leecalcote** [comment]
> One of the more important use cases to facilitate with these logs is that of when a model or any of its entities is not successful in registering. It's important to be able to give contributors and Meshery administrators as much information as possible about what model and what entity failed to import and why.
> 
> Please provide examples of how these logs facilitate the administrator's resolution in the face of these import failures.
> 
> Include a link to troubleshooting Docs?

Be aware that both Meshery Server and its clients (Meshery UI and `mesheryctl`) stand to benefit from this logged information. When a user imports a model using either of these two clients, the clients should return similar information.

**Comment by Jougan-0** [comment]
https://github.com/meshery/meshery/pull/9564

File: models/meshmodel/registry/registry.go (hunk 4 above)
Comment (Jougan-0) [reply models/meshmodel/registry/registry.go]: Removed the old function because we already had a function that return same values

**Review by leecalcote — CHANGES_REQUESTED** [review]
I looked again:

```
meshery git:(regLog) make server
cd server; cd cmd; go mod tidy; \
	BUILD="v0.7.2-rc-1" \
	PROVIDER_BASE_URLS="https://meshery.layer5.io" \
	PORT=9081 \
	DEBUG=true \
	ADAPTER_URLS="localhost:10000 localhost:10001 localhost:10012" \
	APP_PATH="./apps.json" \
	go run main.go error.go;
# command-line-arguments
./main.go:286:7: log.Infof undefined (type "github.com/layer5io/meshkit/logger".Handler has no field or method Infof)
./main.go:287:41: undefined: meshmodel.NonImportModel
./main.go:289:8: log.Errorf undefined (type "github.com/layer5io/meshkit/logger".Handler has no field or method Errorf)
./main.go:294:48: undefined: meshmodel.RegisterAttempts
make: *** [server] Error 1

**Comment by Jougan-0** [comment]
> One of the more important use cases to facilitate with these logs is that of when a model or any of its entities is not successful in registering. It's important to be able to give contributors and Meshery administrators as much information as possible about what model and what entity failed to import and why.
> 
> Please provide examples of how these logs facilitate the administrator's resolution in the face of these import failures.
> 
> Include a link to troubleshooting Docs?

We are using a logic of creating a file of registery_attempts.json that stores every entity that failed 
file is present at
 ```meshery/server/cmd/registery_attempts.json ```

**Comment by Jougan-0** [comment]
> > One of the more important use cases to facilitate with these logs is that of when a model or any of its entities is not successful in registering. It's important to be able to give contributors and Meshery administrators as much information as possible about what model and what entity failed to import and why.
> > Please provide examples of how these logs facilitate the administrator's resolution in the face of these import failures.
> > Include a link to troubleshooting Docs?
> 
> We are using a logic of creating a file of registery_attempts.json that stores every entity that failed file is present at `meshery/server/cmd/registery_attempts.json `

This could be used by the contributors and administrators to resolve the issue related to that entity

**Comment by leecalcote** [comment]
Ready for final review, @Jougan-0?

**Comment by Jougan-0** [comment]
@leecalcote final changes suggested by uzair are left will finish it today

File: models/meshmodel/registry/error.go [hunk 5]
```diff
@@ -1,13 +1,128 @@
...
+			entityCount.Attempt++
+			entityCount.Error = err
+			RegisterAttempts.Policy[entityID] = entityCount
+		} else {
+			RegisterAttempts.Policy[entityID] = EntityErrorCount{Attempt: 1, Error: err}
+		}
+
+		if RegisterAttempts.Policy[entityID].Attempt == 1 {
+			currentValue := NonImportModel[HostnameToPascalCase(h.Hostname)]
+			currentValue.Policies++
```
[the remark this answers is not in the document]
Comment (Jougan-0) [reply models/meshmodel/registry/error.go:119]: Therefor this won't be an issue if the host or h.Hostname is nil if will just store it within the host whose name is ""

[the remark this answers is not in the document]
Comment (VihasMakwana) [reply models/meshmodel/registry/error.go:119]: @Jougan-0 I'm talking about 
`NonImportModel[HostnameToPascalCase(h.Hostname)]`.
What if the value is not present in the map?

[the remark this answers is not in the document]
Comment (VihasMakwana) [reply models/meshmodel/registry/error.go:119]: What if `HostnameToPascalCase(h.Hostname)` is not present in `NonImportModel`

**Comment by VihasMakwana** [comment]
@leecalcote PR looks good, can you take a final look?

**Review by VihasMakwana — APPROVED** [review]

**Review by MUzairS15 — COMMENTED** [review]
Can't we maintain the error and attempt count in the registration funcs itself?

**Comment by Jougan-0** [comment]
> Can't we maintain the error and attempt count in the registration funcs itself?
The registration is done as follows in the 
```https://github.com/meshery/meshery/blob/cfd822700dc4dcb7b4cafa9d3af593fee017cf31/server/meshmodel/helper.go#L163```
so while it is being is registered if we have an error we pass the to the new function I created 
I used this logic as the in components if the schema is empty it returned nil rather than error and for each component and relationship we register it's model so every time we register them if we found an error then we count with the error . 
Hope this makes sense @MUzairS15

**Comment by Jougan-0** [comment]
https://github.com/meshery/meshery/pull/9829

File: models/meshmodel/registry/error.go [hunk 6]
```diff
@@ -1,13 +1,141 @@
...
+}
+func ErrMarshalingRegisteryAttempts(err error) error {
+	return errors.New(ErrMarshalingRegisteryAttemptsCode, errors.Alert, []string{"Error marshaling RegisterAttempts to JSON"}, []string{"Error marshaling RegisterAttempts to JSON: ", err.Error()}, []string{}, []string{})
+}
+func ErrWritingRegisteryAttempts(err error) error {
+	return errors.New(ErrWritingRegisteryAttemptsCode, errors.Alert, []string{"Error writing RegisteryAttempts JSON data to file"}, []string{"Error writing RegisteryAttempts JSON data to file:", err.Error()}, []string{}, []string{})
+}
+func ErrRegisteringEntity(failedMsg string, hostName string) error {
+	return errors.New(ErrRegisteringEntityCode, errors.Alert, []string{fmt.Sprintf("The import process for a registrant %s encountered difficulties,due to which %s. Specific issues during the import process resulted in certain entities not being successfully registered in the table.", hostName, failedMsg)}, []string{fmt.Sprintf("For registrant %s %s", hostName, failedMsg)}, []string{"Could be because of empty schema or some issue with the json or yaml file"}, []string{"Check /server/cmd/registery_attempts.json for futher details"})
```
Comment (leecalcote) [inline models/meshmodel/registry/error.go:64]: @Jougan-0 @MUzairS15 this directory will not be present in Meshery deployments. This log file needs to be stored under `~/.meshery/server/registry.log`

**Review by leecalcote — CHANGES_REQUESTED** [review]
The log file location needs to be available in Meshery deployments, not only in local builds. Our standard location for such files is ~/.meshery

**Pull request closed without merging**